packages/expo-face-detector/ios/EXFaceDetector/EXFaceDetectorManager.m

```diff
@@ -187,76 +216,101 @@ - (long)_getLongOptionValueForKey:(NSString *)key
   return [(NSNumber *)[_faceDetectorOptions valueForKey:key] longValue];
 }
 
-- (void)_updateOptionSettingForKey:(NSString *)key withJSONValue:(NSNumber *)value
-{
-  long requestedValue = [value longValue];
-  long currentValue = [self _getLongOptionValueForKey:key];
-  
-  if (requestedValue != currentValue) {
-    [_faceDetectorOptions setValue:@(requestedValue) forKey:key];
-    [self _runBlockIfQueueIsPresent:^{
-      [self _resetFaceDetector];
-    }];
-  }
-}
-
 - (void)_runBlockIfQueueIsPresent:(void (^)(void))block
 {
   if (_sessionQueue) {
     dispatch_async(_sessionQueue, block);
   }
 }
 
-#pragma mark - GMVDataOutputDelegate
-
-- (void)dataOutput:(GMVDataOutput *)dataOutput didFinishedDetection:(NSArray<__kindof GMVFeature *> *)results
-{
-  // Calling dataOutput:didFinishedDetection with dataOutput that in videoSettings has no information about
-  // width or height started happen after refactor: moving face detection logic from EXCameraManager to EXFaceDetectorManager.
-  // I suppose no information is provided because data output is already disconnected from the input and it has no
-  // information about the source. Let's reset the information then.
-  if (!_connected) {
-    [self _notifyOfFaces:nil];
-    return;
-  }
-  
-  AVCaptureVideoOrientation interfaceVideoOrientation = _previewLayer.connection.videoOrientation;
-  CGAffineTransform transform = [EXFaceDetectorUtils transformFromDeviceOutput:dataOutput toInterfaceVideoOrientation:interfaceVideoOrientation];
+- (void)captureOutput:(AVCaptureVideoDataOutput *)output didOutputSampleBuffer:(CMSampleBufferRef)sampleBuffer fromConnection:(AVCaptureConnection *)connection {
   
-  EXFaceEncoder *faceEncoder = [[EXFaceEncoder alloc] initWithTransform:transform];
+  dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
I have no better idea for that. We need to have update on orientation as fast as possible. We cannot use deviceOrientation, since it is misleading (can be FACE_UP for instance, or can change without interface orientation change). If you have any suggestions, I will gladly accept it.

Reply:
bump

Reply:
That is actually very good idea, will do.